### PR TITLE
Exposed onBan and onTimeout events ( Resolves #23 )

### LINKED ISSUE #23: Listen for bans and timeouts
At the moment, we don't have a nice way of listening for timeout or ban actions that happen in chat. tmi.js exposes these events so it is fairly easy to capture, but means that any code handling bans or timeouts is in a different style to the rest of the event driven code.

https://github.com/tmijs/docs/blob/gh-pages/_posts/v1.4.2/2019-03-03-Events.md#timeout

https://github.com/tmijs/docs/blob/gh-pages/_posts/v1.4.2/2019-03-03-Events.md#ban

I also think it would be nice to add the ban reason from the PubSub feed if the user has supplied a token.

### PR DESCRIPTION
**Pull request by KenAKAFrosty**
These are super helpful for some use cases that we have, but also figured it made sense to send this upstream to help out anyone else who'd also like this functionality! :) ( Also resolves #23 )

### CODE REVIEW

**Comment by instafluff** [comment]
This is awesome, thank you so much! Would you mind also adding it to the README list of events please and I will merge it!

**Comment by KenAKAFrosty** [comment]
> This is awesome, thank you so much! Would you mind also adding it to the README list of events please and I will merge it!

My pleasure! Readme should be all good to go now :D

File: README.md [hunk 1]
```diff
@@ -254,6 +254,12 @@ ComfyJS.Disconnect();
  - **onHosted**`( user, viewers, autohost, extra )`
     - Responds to channel being hosted
     - Requires being authorized as the broadcaster
+ - **onBan**`( bannedUsername, reason, extra )`
+    - Responds to a user being banned
+    - reason will be null unless authorized as broadcaster or moderator for the channel
```
Comment (BenDMyers) [inline README.md:259]: According to <https://github.com/tmijs/docs/blob/gh-pages/_posts/v1.4.2/2019-03-03-Events.md> linked in #23, `reason` is deprecated and should _always_ be `null` - is this still the case?

File: README.md [hunk 2]
```diff
@@ -254,6 +254,12 @@ ComfyJS.Disconnect();
...
+ - **onTimeout**`( timedOutUsername, reason, durationInSeconds, extra )`
+    - Responds to a user being timed out
+    - reason will be null unless authorized as broadcaster or moderator for the channel
```
Comment (BenDMyers) [inline README.md:262]: Same as above - I think `reason` will always be `null`.

Comment (KenAKAFrosty) [reply README.md:262]: Ahh, great catch! I will fix and resubmit. Thank you!!

Comment (KenAKAFrosty) [reply README.md]: Okay pulled out the reason param and README is updated accordingly. Thanks again Ben!

**Comment by instafluff** [comment]
AWESOME!! Merging now, thank you!

**Pull request merged**